js/components/matrix.js

```diff
@@ -0,0 +1,330 @@
+        });
+      }
+    },
+
+    /**
+     * Clears the value for each row
+     * @returns {undefined}
+     */
+    clearValue() {
+      if (!this.$refs.matrix) {
+        return;
+      }
+
+      $(this.$refs.matrix).find('.matrix-radio-input:checked').prop('checked', false);
+
+      if (this.value === '') {
+        this.setDefaultValue();
+      }
+
+      if (typeof this.value === 'string' && this.value !== '') {
+        try {
+          this.value = JSON.parse(this.value);
+        } catch (e) {
+          this.value = '';
+        }
+      }
+    },
+
+    /**
+     * Check if all rows has column values or not
+     * @param {object} val - value of selected options
+     * @returns {String} checkFlag
+     */
+    checkValue: function(val) {
+      if (typeof val === 'string' && val !== '') {
+        try {
+          val = JSON.parse(this.value);
+        } catch (e) {
+          val = '';
+        }
+      }
+
+      var checkFlag = 'set';
+      var $vm = this;
+      var columnOpt = [];
+
+      if (val === '') {
+        val = {};
+        checkFlag = 'clear';
+      } else if (!_.isEmpty(val)) {
+        var result = [];
+
+        _.forIn(val, function(value) {
+          if (typeof value !== 'undefined') {
+            result.push(value);
+          }
+        });
+
+        if (result.length > 0) {
+          _.forEach(result, function(col) {
+            _.find($vm.columnOptions, function(column) {
+              if (column.label === col || column.id === col) {
+                columnOpt.push(column);
+              }
+            });
+          });
+        }
+
+        if (result.length === 0 || columnOpt.length === 0) {
+          checkFlag = 'clear';
+        }
+      }
+
+      return checkFlag;
+    },
+
+    onReset: function(data) {
+      if (data.id === this.$parent.id) {
+        this.clearValue();
+
+        if (this.defaultValueSource !== 'default') {
+          this.setValueFromDefaultSettings({ source: this.defaultValueSource, key: this.defaultValueKey });
+        }
+      }
+    },
+
+    onBeforeSubmit: function(data) {
+      var $vm = this;
+
+      _.forIn(data[this.name], function(key, val) {
+        var row = _.find($vm.rowOptions, function(row) {
+          return (_.has(row, 'label') && _.has(row, 'id')) ? row.id === val : row.label === val;
+        });
+
+        if (!row) {
+          delete data[$vm.name][val];
+        }
+      });
+    },
+
+    /**
+     * Check if the value has correct rowOptions and columnOptions or not
+     * @returns {undefined}
+     */
+    checkInvalidValue: function() {
+      if (!this.value) {
+        return;
+      }
+
+      var validColumns = [];
+      var $vm = this;
+
+      _.forIn(this.value, function(key) {
+        _.find($vm.columnOptions, function(col) {
+          if (col.label === key || col.id === col) {
+            validColumns.push(col);
+          }
+        });
+      });
+
+      return validColumns.length > 0 ? true : false;
```

Comment:
why ternary operator simply `return validColumns.length > 0` will do the work

Reply:
done